src/main/config/config.c

```diff
@@ -229,7 +229,7 @@ static void validateAndFixConfig(void)
     }
 #endif
 
-    if (!isSerialConfigValid(serialConfig())) {
+    if (!isSerialConfigValid(serialConfigMutable())) {
```

Comment:
This was here for 7 years :)

For my understanding we have to a save serialConfig before we can use it. 
Why we need to check the mutable version here?

Reply:
Because as was this could return false and wipe the whole serial config which would annoy me for sure. Now, by passing the mutable version and handling with care it can fix the config without trashing it.

Reply:
Yes annoying indeed. That's make this change worthy of a PR on it self. Good find !!!